corehq/blobs/s3db.py

```diff
@@ -40,6 +41,24 @@ def __init__(self, config):
         # https://github.com/boto/boto3/issues/259
         self.db.meta.client.meta.events.unregister('before-sign.s3', fix_s3_host)
 
+    def report_timing(self, action):
+        blobdb = self
+        timer = TimingContext()
+        original_stop = timer.stop
+
+        def new_stop(timer, name=None):
```

Comment:
I'm guessing `timer` here is conventionally called `self`? I think this is not what you want:

```py
class Context(object):
    def stop(self, name=3):
        print(self, name)

timer = Context()
old_stop = timer.stop

def new_stop(self, name=5):
    old_stop(name)
    print(self, name)

timer.stop = new_stop
timer.stop(7)
```
output:
```
(<__main__.Context object at 0x100789910>, 5)
(7, 5)
```

In other words, this line should say `def new_stop(name=None):`.

Reply:
Thanks, good catch